static/app/views/starfish/queries/useSpanMetricsSeries.tsx

```diff
@@ -78,14 +85,33 @@ function getEventView(
   // TODO: This condition should be enforced everywhere
   // query.addFilterValue('has', 'span.description');
 
+  // Pick the highest possible interval for the given yAxis selection. Find the ideal interval for each function, then choose the largest one. This results in the lowest granularity, but best performance.
+  const interval = sortBy(
+    yAxis.map(yAxisFunctionName => {
+      const parseResult = parseFunction(yAxisFunctionName);
+
+      if (!parseResult) {
+        return DEFAULT_INTERVAL;
+      }
+
+      return getIntervalForMetricFunction(
+        parseResult.name as Aggregate | SpanFunctions,
```

Comment:
l: is the cast safe here? Should we create a helper around `parseFunction` that enforces the type?

Reply:
Not fully safe, no, good point. I think it might be good to type `getIntervalForMetricFunction` as accepting `Aggregate | SpanFunction | string`, which would be more truthful, since it successfully drops to a fallback is the function is not found, rather than creating a helper. Thoughts?

Reply:
I think thats a good option, hopefully makes `getIntervalForMetricFunction` easier to use in other contexts as well.

Reply:
Done!